tests/std/tests/P0040R3_parallel_memory_algorithms/test.cpp

```diff
@@ -0,0 +1,448 @@
+    template <class ExecutionPolicy>
+    void operator()(const size_t testSize, const ExecutionPolicy& exec) {
+        auto buffer         = make_unconstructed_nondestroying_buffer<unsigned char>(testSize);
+        const auto begin_it = buffer.get();
+        const auto end_it   = begin_it + testSize;
+
+        const auto result_it = uninitialized_default_construct_n(exec, begin_it, testSize);
+        assert(end_it == result_it);
+    }
+};
+
+struct test_case_uninitialized_value_construct_parallel {
+    template <class ExecutionPolicy>
+    void operator()(const size_t testSize, const ExecutionPolicy& exec) {
+        auto buffer         = make_unique<unsigned char[]>(testSize);
+        const auto begin_it = reinterpret_cast<wrap_uchar*>(buffer.get());
+        const auto end_it   = begin_it + testSize;
+
+        fill_n(buffer.get(), testSize, bad_uchar);
+
+        uninitialized_value_construct(exec, begin_it, end_it);
+        assert(all_of(begin_it, end_it, expectation));
+    }
+};
+
+struct test_case_uninitialized_value_construct_n_parallel {
+    template <class ExecutionPolicy>
+    void operator()(const size_t testSize, const ExecutionPolicy& exec) {
+        auto buffer         = make_unique<unsigned char[]>(testSize);
+        const auto begin_it = reinterpret_cast<wrap_uchar*>(buffer.get());
+        const auto end_it   = begin_it + testSize;
+
+        fill_n(buffer.get(), testSize, bad_uchar);
+
+        const auto result_it = uninitialized_value_construct_n(exec, begin_it, testSize);
+        assert(all_of(begin_it, end_it, expectation));
+        assert(end_it == result_it);
+    }
+};
+
+struct test_case_uninitialized_value_construct_memset_parallel {
+    template <class ExecutionPolicy>
+    void operator()(const size_t testSize, const ExecutionPolicy& exec) {
+        auto buffer         = make_unique<int[]>(testSize);
+        const auto begin_it = buffer.get();
+        const auto end_it   = begin_it + testSize;
+
+        fill_n(begin_it, testSize, bad_int);
+
+        uninitialized_value_construct(exec, begin_it, end_it);
+        assert(all_of(begin_it, end_it, expectation_zero));
+    }
+};
+
+struct test_case_uninitialized_value_construct_n_memset_parallel {
+    template <class ExecutionPolicy>
+    void operator()(const size_t testSize, const ExecutionPolicy& exec) {
+        auto buffer         = make_unique<int[]>(testSize);
+        const auto begin_it = buffer.get();
+        const auto end_it   = begin_it + testSize;
+
+        fill_n(begin_it, testSize, bad_int);
+
+        const auto result_it = uninitialized_value_construct_n(exec, begin_it, testSize);
+        assert(all_of(begin_it, end_it, expectation_zero));
+        assert(end_it == result_it);
+    }
+};
+
+struct test_case_uninitialized_value_construct_unwrap_parallel {
+    template <class ExecutionPolicy>
+    void operator()(const size_t testSize, const ExecutionPolicy& exec) {
+        auto vec            = vector<int>(testSize, bad_int);

```

Comment:
To other reviewers: I'm not complaining about the use of AAA here, because I appreciate how it increases consistency in these test cases. I prefer `auto x = f(args...);` and `auto y = T(args...);` to `auto x = f(args...);` and `T y(args...);`. YMMV. (Also on 208.) (No change requested.)

Reply:
I'm (reluctantly) okay with this rationale.